### PR TITLE
1513 fix build

### PR DESCRIPTION
**Pull request by victornoel**
This is for #1513.

It bumps cactoos-matchers to latest version and fix build.

I also took the liberty of fixing typing errors that would appear in my IDE but not during the build, as well as simplifying a bit of code while I was at it.

### CODE REVIEW

File: src/main/java/org/cactoos/scalar/MultiplicationOf.java [hunk 1]
```diff
@@ -81,7 +81,7 @@ public MultiplicationOf(final Iterable<? extends Number> src) {
             return new Folded<>(
                 BigDecimal.ONE,
                 (mtn, value) -> mtn.multiply(value, MathContext.DECIMAL128),
-                    new Mapped<>(
+                    new Mapped<BigDecimal>(
```
Comment (andreoss) [inline src/main/java/org/cactoos/scalar/MultiplicationOf.java:84]: @victornoel This issue with type inference here is caused by either older JDK, or IDE version.

Comment (victornoel) [reply src/main/java/org/cactoos/scalar/MultiplicationOf.java:84]: @andreoss yes, it's Eclipse, but it's not incorrect to explicit the type here, so let's keep it, at least people using Eclipse will be able to contribute without bothering with those errors

**Review by andreoss — APPROVED** [review]
@victornoel Seems okay, though the changes are a bit random. 
The type inference issues can be captured by maven-compiler-plugin, or by extra compilation step by ecj.

**Comment by victornoel** [comment]
@andreoss 
> The type inference issues can be captured by maven-compiler-plugin, or by extra compilation step by ecj.

Can you tell me more about the first one?

I know we can use ecj to compile with maven so that we will get the same compiler as in Eclipse and thus the same typing/inference as in Eclipse for consistency (and even maybe improved inference for some cases).
I hesitate to do so just to keep things simpler at cactoos level.

> Seems okay, though the changes are a bit random.

hehe yep, sorry about that, I won't do it again, but there was a bunch of awful things I wanted to fix while at it (independently of the type inference thing).

**Comment by victornoel** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @andreoss[/z](https://www.0crat.com/u/andreoss), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@andreoss please make sure you will find at least 3 issues during next CR, thanks

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**